src/models/message.ts

```diff
@@ -257,23 +229,60 @@ export class MessageModel extends DocumentDbModel<
    */
   public findMessages(
     fiscalCode: FiscalCode
-  ): DocumentDbUtils.IResultIterator<RetrievedMessageWithContent> {
-    return DocumentDbUtils.queryDocuments(
-      this.dbClient,
-      this.collectionUri,
-      {
-        parameters: [
-          {
-            name: "@fiscalCode",
-            value: fiscalCode
-          }
-        ],
-        query: `SELECT * FROM m WHERE m.${MESSAGE_MODEL_PK_FIELD} = @fiscalCode`
-      },
-      fiscalCode
+  ): TaskEither<
+    CosmosErrors,
+    AsyncIterator<ReadonlyArray<t.Validation<RetrievedMessage>>>
+  > {
+    const iterator = this.getQueryIterator({
+      parameters: [
+        {
+          name: "@fiscalCode",
+          value: fiscalCode
+        }
+      ],
+      query: `SELECT * FROM m WHERE m.${MESSAGE_MODEL_PK_FIELD} = @fiscalCode`
+    })[Symbol.asyncIterator]();
+
+    return fromEitherT(
+      tryCatch2v(
+        () => iterator,
+        _ => CosmosErrorResponse(_ as ErrorResponse)
+      )
     );
   }
 
+  /**
+   * @deprecated use getQueryIterator + asyncIterableToArray
```

Comment:
so why leave it?

Reply:
I didn't wanted to change too many things in the functions code - indeed I would rather remove the code, is that ok with you?

Reply:
yes